backends/s3/src/main/scala/geotrellis/s3/S3Client.scala

```diff
@@ -14,10 +14,10 @@ import com.amazonaws.ClientConfiguration
 import org.apache.commons.io.IOUtils
 import scala.collection.JavaConversions._
 
-trait S3Client extends LazyLogging {
+trait S3Client extends LazyLogging with Serializable {
```

Comment:
I see. I would suggesting moving the actual logic to companion object and making parameter of type `() => S3Client`. The companion object can be tested. My feeling has been that `() => S3Client` is sufficiently weird signature that I want to minimize how often it shows up in the API. So I would even say that the extension method need not expose it.